src/main/src/main/java/org/vfny/geoserver/global/GeoServerFeatureSource.java

```diff
@@ -570,7 +572,8 @@ protected Query adaptQuery(Query query, SimpleFeatureType schema) throws IOExcep
         if (newQuery.getCoordinateSystemReproject() != null) {
             newQuery.setCoordinateSystemReproject(null);
         }
-        if (newQuery.getCoordinateSystem() != null) {
+        if (newQuery.getCoordinateSystem() != null
+                && metadata.get(FeatureTypeInfo.OTHER_SRS) == null) {
```

Comment:
this method removes CRS from query, modified it to not do incase of WFS-NG feature type